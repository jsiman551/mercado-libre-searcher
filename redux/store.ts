import { configureStore } from '@reduxjs/toolkit'
import searchDataReducer from './slices/search-data-slice'
import searchInputReducer from './slices/search-input-slice'
import sortOptionReducer from './slices/sort-option-slice'
import sidebarFlagReducer from './slices/sidebar-flag-slice'
import priceRangeReducer from './slices/price-range-slice'

export const store = configureStore({
  reducer: {
    searchData: searchDataReducer,
    searchInput: searchInputReducer,
    sortOption: sortOptionReducer,
    mobileSidebarFlag: sidebarFlagReducer,
    priceRange: priceRangeReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
