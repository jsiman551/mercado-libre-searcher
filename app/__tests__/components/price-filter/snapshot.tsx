import PriceFilter from '@/components/price-filter'
import { store } from '@/redux/store'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'

it('renders price filter component', () => {
  const priceRangeformRef = { current: document.createElement('form') }
  const setPriceRangeSubmit = jest.fn()
  const { container } = render(
    <Provider store={store}>
      <PriceFilter
        priceRangeformRef={priceRangeformRef}
        priceRangeSubmit={false}
        setPriceRangeSubmit={setPriceRangeSubmit}
      />
    </Provider>,
  )
  expect(container).toMatchSnapshot()
})
