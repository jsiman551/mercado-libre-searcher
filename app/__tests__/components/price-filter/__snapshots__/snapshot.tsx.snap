// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders price filter component 1`] = `
<div>
  <div
    class="sc-beyTiQ dntuzr"
    data-testid="priceFilterContainer"
  >
    <h3
      class="sc-guDMob dEpnaU"
    >
      Precio
    </h3>
    <ul
      class="sc-dmyDmy pJniv"
    >
      <li
        class="sc-hLQTFJ fBqJzu"
      >
        <form
          class="sc-iBdoyZ kMxeqU"
        >
          <div
            class="sc-fsYfxw iWgysI"
          >
            <input
              class="sc-qZruQ jNgvfr"
              placeholder="Mínimo"
              type="number"
            />
          </div>
          <div
            class="sc-fsYfxw iWgysI"
          >
            <input
              class="sc-qZruQ jNgvfr"
              placeholder="Máximo"
              type="number"
            />
          </div>
          <div
            class="sc-fsYfxw iWgysI"
          >
            <button
              class="sc-jsEeA-d bgWXgc"
              disabled=""
            >
              <img
                alt="submit price filter"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/arrow-filter-icon.svg"
                style="color: transparent;"
                width="20"
              />
            </button>
          </div>
        </form>
      </li>
    </ul>
  </div>
</div>
`;
